primitives/api/proc-macro/src/impl_runtime_apis.rs

```diff
@@ -424,121 +416,65 @@ fn generate_api_impl_for_runtime(impls: &[ItemImpl]) -> Result<TokenStream> {
 /// with code that calls into the runtime.
 struct ApiRuntimeImplToApiRuntimeApiImpl<'a> {
 	runtime_block: &'a TypePath,
-	runtime_mod_path: &'a Path,
-	runtime_type: &'a Type,
-	trait_generic_arguments: &'a [GenericArgument],
-	impl_trait: &'a Ident,
 }
 
-impl<'a> Fold for ApiRuntimeImplToApiRuntimeApiImpl<'a> {
-	fn fold_type_path(&mut self, input: TypePath) -> TypePath {
-		let new_ty_path =
-			if input == *self.runtime_block { parse_quote!(__SR_API_BLOCK__) } else { input };
-
-		fold::fold_type_path(self, new_ty_path)
-	}
+impl<'a> ApiRuntimeImplToApiRuntimeApiImpl<'a> {
+	/// Process the given item implementation.
+	fn process(mut self, input: ItemImpl) -> ItemImpl {
+		let mut input = self.fold_item_impl(input);
 
-	fn fold_impl_item_method(&mut self, mut input: syn::ImplItemMethod) -> syn::ImplItemMethod {
-		let block = {
-			let runtime_mod_path = self.runtime_mod_path;
-			let runtime = self.runtime_type;
-			let native_call_generator_ident =
-				generate_native_call_generator_fn_name(&input.sig.ident);
-			let call_api_at_call = generate_call_api_at_fn_name(&input.sig.ident);
-			let trait_generic_arguments = self.trait_generic_arguments;
-			let crate_ = generate_crate_access(HIDDEN_INCLUDES_ID);
-
-			// Generate the access to the native parameters
-			let param_tuple_access = if input.sig.inputs.len() == 1 {
-				vec![quote!(p)]
-			} else {
-				input
-					.sig
-					.inputs
-					.iter()
-					.enumerate()
-					.map(|(i, _)| {
-						let i = syn::Index::from(i);
-						quote!( p.#i )
-					})
-					.collect::<Vec<_>>()
-			};
-
-			let (param_types, error) = match extract_parameter_names_types_and_borrows(
-				&input.sig,
-				AllowSelfRefInParameters::No,
-			) {
-				Ok(res) => (
-					res.into_iter()
-						.map(|v| {
-							let ty = v.1;
-							let borrow = v.2;
-							quote!( #borrow #ty )
-						})
-						.collect::<Vec<_>>(),
-					None,
-				),
-				Err(e) => (Vec::new(), Some(e.to_compile_error())),
-			};
+		let crate_ = generate_crate_access(HIDDEN_INCLUDES_ID);
 
-			// Rewrite the input parameters.
-			input.sig.inputs = parse_quote! {
+		input.items.clear();
```

Comment:
```suggestion
		// Delete all functions, because all of them are default implemented by `decl_runtime_apis!`.
		// We only need to implement the `__runtime_api_internal_call_api_at` function.
		input.items.clear();
```